src/Packages.php

```diff
@@ -68,6 +68,16 @@ protected static function load_packages() {
 			}
 			call_user_func( array( $package_class, 'init' ) );
 		}
+
+		// Proxies "activated_plugin" hook for embedded packages listen on WC plugin activation
+		// https://github.com/woocommerce/woocommerce/issues/28697.
+		if ( is_admin() ) {
+			$is_woocommerce_just_activated = get_transient( 'woocommerce_activated_plugin' );
+			if ( $is_woocommerce_just_activated ) {
+				delete_transient( 'woocommerce_activated_plugin' );
+				do_action( 'woocommerce_activated_plugin', 'woocommerce/woocommerce.php' );
```

Comment:
Is it necessary to pass `'woocommerce/woocommerce.php'` here?

Reply:
This is just a quality of life addition, my intention to pass the plugin's filename is to match the signature of `activated_plugin` function, so that the listener function could handle both hooks:

```php
public function __construct() {
  // Handle activation from WooCommerce plugin
  add_action( 'woocommerce_activated_plugin', array( __CLASS__, 'activated_plugin' ) );

  // Handle activation from feature plugin
  add_action( 'activated_plugin', array( __CLASS__, 'activated_plugin' ) );
}

public static function activated_plugin( $plugin ) {
  // Ensure we're only running only on activation hook that originates from our plugin
  if ( explode( '/', plugin_basename( __FILE__ ) )[ 0 ] === explode( '/', $plugin )[ 0 ] ) {
    self::run_once();
  }
}
```

I've changed this to pass the actual plugin filename in 0e5596d through the transient value. If this raises any concerns please let me know!